db/compaction/compaction_iterator.cc

```diff
@@ -88,6 +90,9 @@ CompactionIterator::CompactionIterator(
       merge_out_iter_(merge_helper_),
       blob_garbage_collection_cutoff_file_number_(
           ComputeBlobGarbageCollectionCutoffFileNumber(compaction_.get())),
+      blob_fetcher_(CreateBlobFetcherIfNeeded(compaction_.get())),
+      prefetch_buffers_(
+          CreatePrefetchBufferCollectionIfNeeded(compaction_.get())),
```

Comment:
I am a little concerned about calling these two functions in constructor because they perform dynamic memory allocation. Any chance we move it to somewhere else, e.g. `SeekToFirst()`?

Reply:
I don't think we have to worry about that: the subcompactions' `CompactionIterator`s are also themselves heap-allocated and the constructor already does further allocations (e.g. `CompactionProxy`, a bunch of `vector` members etc.)...